### PR TITLE
KAFKA-17449: Move Quota classes to server-common module

### PR DESCRIPTION
**Pull request by mimaison**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: server-common/src/main/java/org/apache/kafka/server/quota/QuotaUtils.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    /**
+     * Casts provided Measurable to Rate
+     * @throws IllegalArgumentException if given measurable is not Rate
+     */
+    private static Rate measurableAsRate(MetricName name, Measurable measurable) {
+        if (measurable instanceof Rate) {
+            return (Rate) measurable;
+        } else {
+            throw new IllegalArgumentException("Metric $name is not a Rate metric, value " + measurable);
```
Comment (chia7712) [inline server-common/src/main/java/org/apache/kafka/server/quota/QuotaUtils.java]: `$name` does not work in java code.

**Review by chia7712 — COMMENTED** [review]
@mimaison thanks for this nice migration. one small question is left. PTAL

File: server-common/src/main/java/org/apache/kafka/server/quota/QuotaUtils.java (hunk 1 above)
Comment (mimaison) [reply server-common/src/main/java/org/apache/kafka/server/quota/QuotaUtils.java]: Oops, I missed this one!

**Comment by chia7712** [comment]
@mimaison Could you please fix the conflicts :)

**Comment by mimaison** [comment]
Yup, just rebased. Let's wait for the CI to run again

**Review by chia7712 — APPROVED** [review]

**Pull request merged**